incubator/starter/image/Dockerfile-stack

```diff
@@ -60,3 +60,8 @@ WORKDIR /project
 
 # Expose the relevant ports (change this to be specific to your application).
 EXPOSE 8080
+
+# Run as a non-root user in group 0.
+RUN useradd -u 1001 -r -g 0 -s /usr/sbin/nologin -m default \
```

Comment:
It may be better to not specify a uid ie 1001 to the userid you are creating and to let the system assign this, as you do not know what base images people will be using and whether uid 1001 is already being used. You can still do all the operations needed but using the userid ie default, instead of the uid value.

Reply:
This isn't really relevant to the PRs purpose - which is to demonstrate a blocking issue we are having with appsody.

That said, you make a good point. Do you work on the java stacks? This code was essentially copied from https://github.com/appsody/stacks/blob/a3be05c738df79cd99138096c4a21d27778bb50b/incubator/java-spring-boot2/image/Dockerfile-stack#L8 and https://github.com/appsody/stacks/blob/a3be05c738df79cd99138096c4a21d27778bb50b/incubator/java-openliberty/image/Dockerfile-stack#L3  It would help stack authors to have clear guidelines on how to meet the non-root requirement.